lib/please_cli/please_cli/decision_task.py

```diff
@@ -323,6 +344,197 @@ def get_task(task_group_id,
     }
 
 
+NixPath = str
+NixHash = str
+NixHashes = typing.List[typing.Tuple[NixPath, NixHash]]
+# TODO: we need to have more detailed type for Projects but we would need to
+#       have a type for please_cli.config.PROJECTS_CONFIG before that
+Projects = typing.Dict
+
+
+async def read_stream(stream,
+                      log_output: typing.Optional[typing.Callable[[str], None]] = None,
+                      callback: typing.Optional[typing.Callable[[str], None]] = None,
+                      ) -> str:
+    output = []
+    while True:
+        line = await stream.readline()
+        if line:
+            line = line.decode().rstrip('\n')
+            if log_output:
+                log_output(line)
+            output.append(line)
+            if callback:
+                callback(line)
+        else:
+            break
+    return '\n'.join(output)
+
+
+async def run(_command: typing.Union[str, typing.List[str]],
+              stream: bool = False,
+              handle_stream_line: typing.Optional[typing.Callable[[str], None]] = None,
+              log_command: bool = True,
+              log_output: bool = True,
+              secrets: typing.List[str] = [],
+              **kwargs,
+              ) -> typing.Tuple[int, str, str]:
+
+    hide_secrets = cli_common.command.hide_secrets
+
+    command: str
+    if isinstance(_command, str):
+        command = _command
+    else:
+        command = ' ' .join(_command)
+
+    if len(command) == 0:
+        raise click.ClickException('Can\'t run an empty command.')
+
+    _kwargs = dict(
+        stdin=asyncio.subprocess.DEVNULL,  # no interactions
+        stdout=asyncio.subprocess.PIPE,
+        stderr=asyncio.subprocess.PIPE,
+    )
+    _kwargs.update(kwargs)
+
+    if log_command:
+        log.debug('Running command', command=hide_secrets(command, secrets), kwargs=_kwargs)
+
+    process = await asyncio.create_subprocess_shell(command, **_kwargs)  # noqa
+
+    if stream:
+        _log_output: typing.Optional[typing.Callable[[str], None]] = None
```

Comment:
put in `else` case to make it more explicit, especially as it looks a lot like `log_output`